cc/imgproc/imgproc.cc

```diff
@@ -128,85 +110,6 @@ NAN_METHOD(Imgproc::GetPerspectiveTransform) {
   info.GetReturnValue().Set(Mat::Converter::wrap(cv::getPerspectiveTransform(srcPoints, dstPoints)));
 }
 
-NAN_METHOD(Imgproc::CalcHist) {
-	FF::TryCatch tryCatch("Imgproc::CalcHist");
-  cv::Mat img, mask = cv::noArray().getMat();
-  std::vector<std::vector<float>> _ranges;
-  if (
-	  Mat::Converter::arg(0, &img, info) ||
-	  Mat::Converter::optArg(2, &mask, info)
-	) {
-	  return tryCatch.reThrow();
-  }
-  if (!info[1]->IsArray()) {
-	  return tryCatch.throwError("expected arg 1 to be an array");
-  }
-  v8::Local<v8::Array> jsHistAxes = v8::Local<v8::Array>::Cast(info[1]);
-
-  cv::Mat inputImg = img;
-  int inputType = CV_MAKETYPE(CV_32F, img.channels());
-  if (inputType != img.type()) {
-    img.convertTo(inputImg, inputType);
-  }
-
-  int dims = jsHistAxes->Length();
-  int* channels = new int[dims];
-  int* histSize = new int[dims];
-  std::vector<float*> ranges;
-  // TODO replace old macros
-  for (int i = 0; i < dims; ++i) {
-    ranges.push_back(new float[dims]);
-    v8::Local<v8::Object> jsAxis = Nan::To<v8::Object>((Nan::Get(jsHistAxes, i).ToLocalChecked())).ToLocalChecked();
-	if (!FF::hasOwnProperty(jsAxis, "ranges")) {
-		return tryCatch.throwError("expected axis object to have ranges property");
-	}
-	v8::Local<v8::Value> jsRangesVal = Nan::Get(jsAxis, Nan::New("ranges").ToLocalChecked()).ToLocalChecked();
-	if (!jsRangesVal->IsArray()) {
-		return tryCatch.throwError("expected ranges to be an array");
-	}
-	v8::Local<v8::Array> jsRanges = v8::Local<v8::Array>::Cast(jsRangesVal);
-	if (jsRanges->Length() != 2) {
-		return tryCatch.throwError("expected ranges to be an array of length 2");
-	}
-    ranges.at(i)[0] = FF::DoubleConverter::unwrapUnchecked(Nan::Get(jsRanges, 0).ToLocalChecked());
-    ranges.at(i)[1] = FF::DoubleConverter::unwrapUnchecked(Nan::Get(jsRanges, 1).ToLocalChecked());
-    int channel, bins;
-
-	if (FF::IntConverter::prop(&channel, "channel", jsAxis) || FF::IntConverter::prop(&bins, "bins", jsAxis)) {
-		return tryCatch.reThrow();
-	}
-    channels[i] = channel;
-    histSize[i] = bins;
-  }
-
-  cv::MatND hist;
-  if (dims == 1) {
-    hist = calcHist1(inputImg, mask, channels, histSize, ranges);
-  }
-  else if (dims == 2) {
-    hist = calcHist2(inputImg, mask, channels, histSize, ranges);
-  }
-  else if (dims == 3) {
-    hist = calcHist3(inputImg, mask, channels, histSize, ranges);
-  }
-  else if (dims == 4) {
-    hist = calcHist4(inputImg, mask, channels, histSize, ranges);
-  }
-
-  for (int i = 0; i < dims; ++i) {
-    delete[] ranges.at(i);
-  }
-  delete[] channels;
-  delete[] histSize;
-
-  int outputType = CV_MAKETYPE(CV_64F, img.channels());
-  if (outputType != hist.type()) {
-    hist.convertTo(hist, outputType);
-  }
```

Comment:
I'll take this opportunity to ask something sort-of on topic that i've always wondered 
Why converting to `64F`?
We use `compareHistAsync` right after `calcHistAsync` and it needs `32F` so we have to add another step to change it back to `32F`

Reply:
I intended to never return 32F Mats to javascript since numbers in javascript are 64 bit. I found obtaining float values from a 32F Mat in javascript will yield slightly different values.